### ISSUE TITLE
about how to have naming self-checks counters differently for other events but remain as the same counter at waiting event.

### ISSUE THREAD

**Issue by WILLSONXWX** [nus-cs2030/2223-s2#720]
Currently, what I have is a class called waitingArea that extends server and contains a number of self-check counters. By naming my WaitingArea as self-check 2, I managed to have every output as self-check 2. However, I just realised that except for wait event, I should have different self-check counters doing it. May I know how to achieve that?

**Comment by sethteo** [comment]
You can try having your self-check counters extend the server instead and your waitingArea just be the one the manages the self-check counters within it

**Comment by ayevxn** [comment]
In my implementation, both the waitingArea and self-check counters extend Server. To identify which self-check counter was serving a particular customer, I defined a new getIndex(Customer customer) method in self-check counter which iterated through the self-check counters and returned the counter that was 'holding' the given customer. In the Server class, this method simply returns the index of the Server.

**Comment by cheng-yee** [comment]
Use polymorphism.  When your Serve event wants to generate a string, it should call a method in server to return its identity (server # or self-check #). If the Serve event calls this method in a human server, the human server should return server #. If the Serve event calls this method in WaitingArea, WaitingArea needs to figure out which check-out counter will serve the customer and return the correct self-check #.

**Comment by kyledilao777** [comment]
yep, self checkout has an extended logic to server. you should use self checkout as a subclass of server

**Comment by WILLSONXWX** [comment]
thks guys, I manage to solve this issue by putting in the information of customers in my self checkout counters to identify it
